contracts/VaultV3.vy

```diff
@@ -612,6 +612,7 @@ def _process_report(strategy: address) -> (uint256, uint256):
         # if fee manager is not set, fees are zero
         if fee_manager != ZERO_ADDRESS:
             total_fees = IFeeManager(fee_manager).assess_fees(strategy, gain)
+            assert total_fees <= gain, "fees greater than gain"
```

Comment:
this looks opinionated. 

I'd rather leave the door open for fees to be higher than gains. We don't know if someone will be happy to charge fees in a different way. 

Replace this with a check in the if-else (if total_fees > gain you can skip some profit buffer calculations)

Reply:
I´ve added logic and tests to handle FeeManagers that might have fees greater or equal to gain.